README.md

```diff
@@ -1,132 +1,181 @@
+sought by bumping the RFC to the top of the RFC table on the Concourse website.
+No further changes should be made to the proposal during this period.
+
+If there is a challenge to the resolution during the quiet period the label may
+be removed at the discretion of the assignee, and the RFC process will continue
+as before.
 
 
 ## Implementing an RFC
 
-Once an RFC is accepted, an associated issue will be opened on the [`concourse`
-repository](https://github.com/concourse/concourse) repository, referencing the
-RFC's pull request. This issue can be created by the RFC author or assignee.
+When an RFC is merged the core team assignee is responsible for opening an
+issue on the [Concourse repository](https://github.com/concourse/concourse) to
+keep track of its implementation. The issue can be lightweight and just
+reference the RFC. The assignee must also add a link to the issue at the top of
+the RFC's proposal document.
+
+The [**maintainers** team][maintainers-team] is responsible for determining the
+proposal's priority by adding a **priority/high**, **priority/medium**, or
+**priority/low** label to the RFC's issue. Priority is an approximation of
+overall value and desired timeline for implementation.
 
-By the time an RFC is merged, we should have a pretty good idea of who's going
-to implement it. This may or may not be the same person that submitted the RFC.
-Large-scale proposals for example may be picked up by the core Concourse team
-instead (but obviously that'd be something we agree on prior to merging).
+An RFC author is not necessarily responsible for its implementation, though
+they may volunteer. If the maintainers have sufficient bandwidth they may place
+it on their roadmap by prioritizing the issue in a GitHub project. Otherwise
+the maintainers will add a **help wanted** label to the issue.
 
-The implementation process itself falls under the normal [Concourse development
-process](https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md).
+In any case, contributors may volunteer to implement a proposal provided that
+work has not already begun. If you would like to volunteer, please leave a
+comment on the issue to let others know!
+
+From there, the implementation process falls under the normal [Concourse
+development process][contributing].
+
+
+## Revising an RFC
+
+RFCs represent the planning phase. An RFC's proposal is not the source of truth
+for the feature's documentation, and should not be revised to keep up with
+later iterations after the initial proposal is implemented. A new RFC should be
+proposed for subsequent changes instead.
+
+If an RFC is merged and later changes are deemed necessary prior to final (i.e.
+non-experimental) implementation, a follow-up PR may be submitted that updates
+the proposal in-place. In this case the RFC author must include a MAJOR.MINOR
+revision number in the proposal and maintain a brief summary of changes at the
+bottom of the proposal.
```

Comment:
Is there a procedure for deprecating / obsoleting an RFC? Let's say `spatial-resources` was merged in as an RFC, but then we came out with `resource-prototypes`, how to we mark it as "superseded" (or "partially superseded")? And who's responsibility is it to make the change to the old RFC (I'm guessing it's the new RFC's author)?

Reply:
I think I should've created this comment on a newline instead of on that particular paragraph, but I was more concerned with what's the procedure for after RFC is merged and the final implementation is done. Let's say `040-valid-identifiers` is merged (which it currently is), and has gone GA'd (which it currently isn't).

1. Is there a procedure for updating the proposal (e.g. `~` is now a valid character)? Does it need to go through the experimental implementation phase once again? Or maybe do we just submit a *new* RFC and mark 040-valid-identifiers as `replaced by: ###-valid-identifiers-v2`?
2. What happens if there's a new RFC that partially replaces it? Let's say we have a new RFC `###-valid-job-syntax` which overlaps (e.g. stricter identifiers for jobs) but also contains additional content (e.g. reserved words, structure, etc). What's the wording for "Some part of this RFC has been supplemented by RFC ###"?

Reply:
Those are good questions, but I think they may indicate that we have two different perspectives of RFCs getting tangled up between our comments:

* RFCs serve as a collaboration tool for planning changes to a single project.
* RFCs serve as authoritative design documents for the broader ecosystem to rally around and implement (e.g. open standards and protocols like SSH).

Both are perfectly valid but serve different use cases. What I'm trying to say is that our RFC process is currently designed for the former, not the latter. We have quite a few proposals that seem like the latter, but that's because they introduce entirely new concepts (e.g. prototypes, projects) or define internal standards for the first time (e.g. valid identifiers). I expect later iterations to be done through smaller followup RFCs, and for the Concourse documentation to remain the authoritative description of the current state of the world.

For example, imagine someone joins the project 5 years from now and wants to make `*` valid in identifiers. I wouldn't expect them to go through all previous RFCs to find anything that mentions identifiers so they can reference it. I also wouldn't expect them to introduce a new 'valid identifiers' proposal to replace the old one, because that would make it harder for reviewers to know what changes are actually being proposed (i.e. what's the delta).

So, if the `040-valid-identifiers` RFC was completed and GA'd, it would then just become part of the baseline for any future RFCs, and the `040-valid-identifiers` document should be effectively frozen. Any followup changes to identifiers would need to be submitted as a new proposal, and the new proposal wouldn't even need to be aware of the previous one.

The RFC author may choose to link to prior work in the motivation/summary/context/etc., but I don't think the reverse is necessary, and I don't think we need to encode a process for either. Over time managing these links may lead to high overhead, and it's likely for connections to be missed eventually as older proposals become common assumptions, making the whole process kind of moot.

Reviewers may bring up overlap with other proposals and request clarification, but I don't know if we need a special procedure for that either - there are probably lots of things reviewers might bring up, and I'd prefer to err on the side of being too lightweight rather than too prescriptive, at least until we see how things play out.

Let me know if this makes sense!

Reply:
👍  I was indeed thinking of RFCs more like the IETF living-standards style